libavpipe/src/avpipe_xc.c

```diff
@@ -1802,8 +1860,8 @@ set_idr_frame_key_flag(
         if (frame->pts >= encoder_context->last_key_frame + params->video_seg_duration_ts) {
             int64_t diff = frame->pts - (encoder_context->last_key_frame + params->video_seg_duration_ts);
             int missing_frames = 0;
-            /* We can have some missing_frames only when transcoding UDP MPEG-TS */
-            if (encoder_context->is_mpegts && encoder_context->calculated_frame_duration > 0)
+            /* We can have some missing_frames only when transcoding a UDP live source */
+            if (is_live_source_udp(encoder_context) && encoder_context->calculated_frame_duration > 0)
```

Comment:
Can this also happen with RTP, or no? Seems like no from a brief glance, but I'm not too familiar